### PR TITLE
Implement delete and recover LROs on CertificateClient

### LINKED ISSUE #8009: Deleting Certificates should be an LRO
Since `DeleteCertificate` can take a while to actually work (i.e. the method returns, but the operation doesn't complete right away), we should change this to a long-running operation, but should probably call it `StartDeleteCertificate`. This name does mean we could do it later if decided across languages.

### PR DESCRIPTION
**Pull request by heaths**
Fixes #8009

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/README.md [hunk 1]
```diff
@@ -86,25 +97,24 @@ can also manage certificate issuers, contacts, and management policies of certif
 illustrated in the examples below.
 
 ## Examples
-The Azure.Security.KeyVault.Certificates package supports synchronous and asynchronous APIs.
+The Azure.Security.KeyVault.Secrets package supports synchronous and asynchronous APIs.
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/README.md]: Certificates

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs [hunk 2]
```diff
@@ -433,18 +436,19 @@ public virtual Response<KeyVaultCertificateWithPolicy> RecoverDeletedCertificate
         /// </summary>
         /// <param name="certificateName">The name of the <see cref="DeletedCertificate"/></param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <returns>The recovered certificate and policy</returns>
-        public virtual async Task<Response<KeyVaultCertificateWithPolicy>> RecoverDeletedCertificateAsync(string certificateName, CancellationToken cancellationToken = default)
+        /// <returns>A <see cref="RecoverDeletedCertificateOperation"/> to wait on this long-running operation.</returns>
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs]: Can we add add a comment to these, which states that you don't have to wait? not sure how to word it.

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/DeleteCertificateOperation.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+                case 404:
+                    return false;
+
+                default:
+                    throw response.CreateRequestFailedException();
+            }
+        }
+    }
+}
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/src/DeleteCertificateOperation.cs]: Could possibly simplify this by adding one Async method with an isAsync parameter.

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/RecoverDeletedCertificateOperation.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+        /// <inheritdoc />
+        public override ValueTask<Response<KeyVaultCertificateWithPolicy>> WaitForCompletionAsync(CancellationToken cancellationToken = default) =>
+            this.DefaultWaitForCompletionAsync(s_defaultPollingInterval, cancellationToken);
+
+        /// <inheritdoc />
+        public override ValueTask<Response<KeyVaultCertificateWithPolicy>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken) =>
+            this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+
+        private static async Task<bool> CheckCompletedAsync(Response response)
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/src/RecoverDeletedCertificateOperation.cs]: same comment as in the delete certificate poller.

**Review by schaabs — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs (hunk 2 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs]: Sure. In hindsight, you had me do that for Keys and Secrets so I can just use that text.

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/DeleteCertificateOperation.cs (hunk 3 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/DeleteCertificateOperation.cs]: One returns a task. I suppose I could change that to a `ValueTask<bool>` but this is so small I'm not sure even that overhead is worth it.

**Pull request merged**